python/ray/workflow/common.py

```diff
@@ -110,31 +83,23 @@ class WorkflowStaticRef:
     """
 
     # The ID of the step that produces the output of the workflow.
-    step_id: StepID
-    # The ObjectRef of the output.
-    ref: ObjectRef
-    # This tag indicates we should resolve the workflow like an ObjectRef, when
-    # included in the arguments of another workflow.
-    _resolve_like_object_ref_in_args: bool = False
+    task_id: StepID
+    # The ObjectRef of the output. If it is "None", then the output has been
+    # saved in the storage, and we need to check the workflow management actor
+    # for the object ref.
+    ref: Optional[ObjectRef] = None
```

Comment:
Does this mean it has some dependence here for this two fields? From the comment it's not very clear. For this structure, I'd prefer making it decoupled.

Can we make `ref` as a cache for task_id? We probably can have another function like get_object to return a ObjectRef for this output.

Reply:
covered by the discussion last week